src/Blocks/BlockPiston.cpp

```diff
@@ -192,39 +268,43 @@ void cBlockPistonHandler::RetractPiston(int a_BlockX, int a_BlockY, int a_BlockZ
 		return;
 	}
 
+	Vector3i pushDir = MetadataToOffset(pistonMeta);
+
 	// Check the extension:
-	AddPistonDir(a_BlockX, a_BlockY, a_BlockZ, pistonMeta, 1);
-	if (a_World->GetBlock(a_BlockX, a_BlockY, a_BlockZ) != E_BLOCK_PISTON_EXTENSION)
+	if (a_World->GetBlock(a_BlockX + pushDir.x, a_BlockY + pushDir.y, a_BlockZ + pushDir.z) != E_BLOCK_PISTON_EXTENSION)
```

Comment:
Theres already an overload of getBlock to use vectors, I was referring to RetractPiston